L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -108,247 +108,324 @@ void PurgeDuplicate::execute(std::vector<Track>& outputtracks_, unsigned int iSe
+              continue;
+            Tracklet* aTrack = inputtrackfits_[i]->getTrack(j);
+            inputtracklets_.push_back(inputtrackfits_[i]->getTrack(j));
+            std::vector<const Stub*> stublist = inputtrackfits_[i]->getStublist(j);
+            inputstublists_.push_back(stublist);
+            std::vector<std::pair<int, int>> stubidslist = inputtrackfits_[i]->getStubidslist(j);
+            inputstubidslists_.push_back(stubidslist);
+            mergedstubidslists_.push_back(stubidslist);
+
+            // Encoding: L1L2=0, L2L3=1, L3L4=2, L5L6=3, D1D2=4, D3D4=5, L1D1=6, L2D1=7
+            // Best Guess:          L1L2 > L1D1 > L2L3 > L2D1 > D1D2 > L3L4 > L5L6 > D3D4
+            // Best Rank:           L1L2 > L3L4 > D3D4 > D1D2 > L2L3 > L2D1 > L5L6 > L1D1
+            // Rank-Informed Guess: L1L2 > L3L4 > L1D1 > L2L3 > L2D1 > D1D2 > L5L6 > D3D4
+            unsigned int curSeed = aTrack->seedIndex();
+            std::vector<int> ranks{1, 5, 2, 7, 4, 3, 8, 6};
+            if (settings_.extended())
+              seedRank.push_back(9);
+            else
+              seedRank.push_back(ranks[curSeed]);
+
+            if (stublist.size() != stubidslist.size())
+              throw "Number of stubs and stubids don't match up!";
+
+            trackInfo.emplace_back(i, false);
+            trackBinInfo.emplace_back(false);
+          } else
+            continue;
         }
+      }
 
-        if (stublist.size() != stubidslist.size())
-          throw "Number of stubs and stubids don't match up!";
+      if (inputtracklets_.empty())
+        continue;
+      unsigned int numStublists = inputstublists_.size();
 
-        trackInfo.emplace_back(i, false);
+      if (settings_.inventStubs()) {
+        for (unsigned int itrk = 0; itrk < numStublists; itrk++) {
+          inputstublists_[itrk] = getInventedSeedingStub(iSector, inputtracklets_[itrk], inputstublists_[itrk]);
+        }
       }
-    }
 
-    if (inputtracklets_.empty())
-      return;
-    unsigned int numStublists = inputstublists_.size();
+      // Initialize all-false 2D array of tracks being duplicates to other tracks
+      bool dupMap[numStublists][numStublists];  // Ends up symmetric
+      for (unsigned int itrk = 0; itrk < numStublists; itrk++) {
+        for (unsigned int jtrk = 0; jtrk < numStublists; jtrk++) {
+          dupMap[itrk][jtrk] = false;
+        }
+      }
 
-    // Initialize all-false 2D array of tracks being duplicates to other tracks
-    bool dupMap[numStublists][numStublists];  // Ends up symmetric
-    for (unsigned int itrk = 0; itrk < numStublists; itrk++) {
-      for (unsigned int jtrk = 0; jtrk < numStublists; jtrk++) {
-        dupMap[itrk][jtrk] = false;
+      // Used to check if a track is in two bins, is not a duplicate in either bin, so is sent out twice
+      bool noMerge[numStublists];
+      for (unsigned int itrk = 0; itrk < numStublists; itrk++) {
+        noMerge[itrk] = false;
       }
```

Comment:
```suggestion
      bool noMerge[numStublists] = {false};
```

Reply:
I believe your C++ statement says to initialize the1st element of the array to "false" and all other elements to their default value (which happens to be "false" too). So it could presumably be simplified to "bool noMerge[numStublists]= {}. But is it considered good style to use the default value rather than initializing the array explicitely?

Reply:
I won't insist on it: mine was just a suggestion to simplify the code and make it more easily readable. It was certainly not a relevant comment, indeed